pygmt/datasets/samples.py

```diff
@@ -68,28 +66,22 @@ def load_sample_data(name):
     if name not in names:
         raise GMTInvalidInput(f"Invalid dataset name '{name}'.")
 
-    # Dictionary of public load functions for backwards compatibility
-    load_func_old = {
-        "mars_shape": load_mars_shape,
-    }
-
     # Dictionary of private load functions
     load_func = {
         "bathymetry": _load_baja_california_bathymetry,
         "earth_relief_holes": _load_earth_relief_holes,
         "fractures": _load_fractures_compilation,
         "hotspots": _load_hotspots,
         "japan_quakes": _load_japan_quakes,
+        "mars_shape": _load_mars_shape,
         "maunaloa_co2": _load_maunaloa_co2,
         "notre_dame_topography": _load_notre_dame_topography,
         "ocean_ridge_points": _load_ocean_ridge_points,
         "rock_compositions": _load_rock_sample_compositions,
         "usgs_quakes": _load_usgs_quakes,
     }
 
-    if name in load_func_old:
-        data = load_func_old[name](suppress_warning=True)
-    elif name in load_func:
+    if name in load_func:
```

Comment:
We already checked if a valid name is given at line 66, so no need to check `if name in load_func`.

Please change:
```
if name in load_func:
    data = load_func[name]()
return data
```
to
```
return load_func[name]()
```